.github/workflows/github_action_test_suite.yml

```diff
@@ -5,188 +5,137 @@ name: test-suite
 
 on:
   push:
-    branches: [ master, develop ]
+    branches: [master, develop]
     paths-ignore:
-      - 'docs/**'
+      - "docs/**"
   pull_request:
-    branches: [ master, develop ]
+    branches: [master, develop]
 
 jobs:
-  build:
+  test:
+    name: Test
     runs-on: ubuntu-latest
 
     strategy:
       fail-fast: false
       matrix:
-        python-version: ['3.9', '3.10', '3.11']
-        TESTING_DB: ['sqlite3', 'postgresql', 'mysql']
+        python-version: ["3.9", "3.10", "3.11"]
+        TESTING_DB: ["sqlite3", "postgresql", "mysql"]
+        include:
+          - python-version: "3.10"
```

Comment:
What is this extra `python-version` for?

Reply:
TL;DR: It's to specify which run will run coverage testing.  When the matrix strategy currently being employed is `python-version: 3.10` and `TESTING_DB: sqlite3`, the variable `coverage-test` will be "included" and set to `true`.

Longer version: you can "include" additions to your matrix configuration variables without forcing the addition to be a part of every single run.  When you use `include` in your matrix, you're telling the job runner that you want to add extra and/or specific configurations to your run.  In this case, I want to flag a specific run (3.10/sqlite3 as you designated) as the test which runs `coverage` and submits the results.  You did this previously by hardcoding it in the step's if condition: `if: ${{ matrix.python-version == "3.10" && matrix.TESTING_DB == "sqlite3" }}`.  I took out the hardcoded conditions and swapped them for a variable designating a run as a coverage-tested run, which is true only when we include it.

In layman's terms, what I did was say, add `coverage-test` (accessible as `${{ matrix.coverage-test }}`) with a value of true, and the run I'd like to add it to has `python-version` as "3.10" and `TESTING_DB` "sqlite3".  Since 3.10 and sqlite3 exist as part of the existing strategy, it simply inserts `coverage-test` into that run rather than create a new run with that variable, thus avoiding redundancy.  The documentation doesn't explicitly say anything about "non-existent" variables when outside that specific run, but I refer you to the below link which references the variable outside that specific run without any apparent issue.

Compare with this example from GitHub Docs: https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#example-expanding-configurations

To adapt the wording from the official documentation:

> When the job for the `python-version` value of `3.10` and `TESTING_DB` value of `sqlite3` runs, an additional variable called `coverage-test` with the value of `true` will be included in the job.

Reply:
Thanks for the explanation, was not familiar with this syntax :+1: